### ISSUE TITLE
how to hot-import config?

### ISSUE THREAD

**Issue by suntong** [wechaty/wechaty-getting-started#56]
I'm trying to [hot-import config](https://github.com/suntong/nodecl/blob/8ddf5eb69d8d84a64d8f26118358034f2716de4b/wx/hot-import-bot/on-message.js#L21-L26):

```js
const { hotImport } = require('hot-import')

async function onMessage (message) {
  const cfg = await hotImport('./config')
  console.log(`Received message: ${message} from ${cfg.db.host}`)
}
```

It works for the first received message. However on receiving the 2nd message, the program errored out with:

```
(node:80803) UnhandledPromiseRejectionWarning: TypeError: Cannot read property 'db' of undefined
    at Wechaty.onMessage (/path/to/wx/hot-import-bot/on-message.js:25:56)
(node:80803) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
```

How can I make it works please?

**Comment by suntong** [comment]
Made [another attempt](https://github.com/suntong/nodecl/blob/0349d3ff27d90adfea0c4e13fa8301622b4e0c74/wx/hot-import-bot/on-message.js) to fix it myself, still nok

**Comment by zixia** [comment]
Hi, I did not use a constant in hotMod before, however, I had just checked the document and it seems should support it. (at <https://github.com/zixia/hot-import> )

Can you try to run the example at <https://github.com/zixia/hot-import#full-example> ? the source is at <https://github.com/zixia/hot-import/blob/master/examples/demo-js.ts>

I'll also have a look into your code(which use a constant in module like `cfg.db.host`) later.

**Comment by suntong** [comment]
Yeah, the demo-js.ts works fine for me, I've tested it as

- [demo1.js](https://github.com/suntong/nodecl/blob/master/sbaa/cfg_hot/demo1.js)
- [demo2.js](https://github.com/suntong/nodecl/blob/master/sbaa/cfg_hot/demo2.js)

both work fine.

**Comment by suntong** [comment]
I think the problem is not constant in hotMod, but the `await hotImport` in `async function onMessage` that is causing the problem. 

Once again, It works for the _first_ received message but not the _second_.

**Comment by huan** [comment]
Did you find any way to resolve this issue and could you share it for us? Thanks.

**Comment by suntong** [comment]
> I think ... the **await** hotImport in **async** function onMessage that is causing the problem.

Moving the `await hotImport` out of the async onMessage event-handling function solved the problem, because the **_redefinition_** was indeed the problem.

**Comment by huan** [comment]
Yes, we should only use `hotImport` once for a module from the beginning of the code.

Glad to hear from you that this issue had been fixed. Cheers and happy Christmas!

**Comment by suntong** [comment]
Oh, actually it is _**not**_ working -- It _seems_ working because I have a `let cfg = require('./config')` at the top of the script, but I was _not_ able to make it hotImported. 

Reopening it, and hope someone can put hot-import config into the demo of https://github.com/wechaty/wechaty-getting-started/tree/master/examples/professional/hot-import-bot

It isn't hard for me _**if**_ I know nodejs well. 

thx

**Comment by mkliu** [comment]
The same problem happens to me too, if it's imported twice there's a problem. The error repros on docker

**Comment by mkliu** [comment]
I changed to this

    if(!config)
    {
      config = await hotImport('./config.js')
    }
